src/Multilog/Multilog.php

```diff
@@ -67,43 +80,20 @@ public function c($name)
      */
     private function initLoggers(array $loggers)
     {
-        foreach ($loggers as $channel => $config) {
-            $this->createLogger($channel, $config);
+        foreach ($loggers as $channel => $closure) {
+            $this->createLogger($channel, $closure);
         }
     }
 
     /**
-     * Create new logger instance with given
-     * configuration and store it in array
-     * with channel name
+     * Store the logger created via closure in with the given channel name
      *
      * @param  string $channel Channel name
-     * @param  array $config  Channel configuration
+     * @param  \Closure $closure  Logger instantiator
      * @return void
      */
-    private function createLogger($channel, array $config)
+    private function createLogger($channel, \Closure $closure)
     {
-        // Setup configuration
-        // Use default laravel logs path
-        $storagePath = $this->app->make('path.storage');
-        $filepath = $storagePath . '/logs/' . $config['stream'];
-
-        $logger = new Logger($channel);
-        $handler = new StreamHandler($filepath);
-
-        // Daily rotation
-        if ($config['daily']) {
-            $handler = new RotatingFileHandler($filepath, 1);
-        }
-
-        // Format line
-        if (isset($config['format'])) {
-            $format = $config['format'];
-            $handler->setFormatter(new LineFormatter($format['output'], $format['date']));
-        }
-
-        $logger->pushHandler($handler);
-
-        $this->channels[$channel] = $logger;
+        $this->channels[$channel] = $closure($channel);
```

Comment:
instead of forcing to a closure, pass as a parameter, but delegate to a handler.

```php
$this->channels[$channel] = $config instanceof \Closure ? $config($channel) : $this->handle($config);
```